applications/app/controllers/ReaderRevenueController.scala

```diff
@@ -0,0 +1,32 @@
+package controllers
+
+import common.{ImplicitControllerExecutionContext, Logging}
+import model._
+import play.api.mvc.{Action, AnyContent, BaseController, ControllerComponents}
+import services.S3
+
+import scala.concurrent.duration._
+import conf.Configuration.readerRevenue._
+import model.Cached.{RevalidatableResult}
+
+
+class ReaderRevenueController(val controllerComponents: ControllerComponents)(implicit context: ApplicationContext)
+  extends BaseController with ImplicitControllerExecutionContext with Logging {
+
+  private def getContributionsBannerDeployLog(): Option[String] = {
+    S3.get(contributionsBannerDeployLogKey)
+  }
+
+  private def bannerDeployLogUnavailable() = {
+    log.warn(s"Could not get reader revenue contributions-banner deploy log from s3")
+    NoCache(NotFound)
+  }
+
+  def contributionsBannerDeployLog(): Action[AnyContent] = Action { implicit request =>
+    getContributionsBannerDeployLog.fold(bannerDeployLogUnavailable){ bannerDeployLog =>
+      Cached(5.minutes) {
```

Comment:
You could possibly do the same as the fronts tool here and just cache it 'forever' - maybe a week or something - assuming that you'll be updating this field via the admin app then that app has the power to clear the fastly cache anyway, so you can just clear the cache when this field gets updated.